plugins/services/src/js/containers/tasks/TasksContainer.js

```diff
@@ -27,20 +41,31 @@ class TasksContainer extends React.Component {
 
     this.state = {
       actionErrors: {},
-      pendingActions: {}
+      pendingActions: {},
+      filterExpression: new DSLExpression(""),
+      filters: new DSLFilterList([]),
+      defaultFilterData: { zones: [], regions: [] }
     };
 
     METHODS_TO_BIND.forEach(method => {
       this[method] = this[method].bind(this);
     });
   }
 
+  componentWillMount() {
+    this.propsToState(this.props);
```

Comment:
https://twitter.com/dan_abramov/status/749710501916139520?lang=en 😏 .

I think this should just be calculated in the render method.

Reply:
I agree with this in general, but in this case I think it is better to keep the filter and defaultFilterData in state. When they don't change from the props we want the *exact* same object, if they change this will trigger => https://github.com/dcos/dcos-ui/blob/master/src/js/components/DSLFilterField.js#L56 and reset everything.

Reply:
Makes sense. So they are stored in state because you want to compare with the previous values, and decide if you update the `filters` and `defaultFilterData` triggering a full re-render. I agree, this is good to keep 👍

Reply:
💭  Can't we achieve the same without the need to store this in state by using `shouldComponentUpdate`?

Reply:
@orlandohohmeier I'm interested only in the filter changes. We receive from the properties the tasks and various other information.
So, if we receive some other tasks with the same region, zones I would like to update this component but keep the filter the same. If we return false when the filter doesn't change then it will become inactive and won't refresh with the new tasks. Also, it wouldn't be correct I think to compare the tasks and all of the other properties in the shouldComponentUpdate, it would become a bit complex.

Reply:
Sounds like a data/control flow issues which we can fix by adjusting the components, token and filter mechanism. Anyway, let's move this implementation forward and create an issue so that we don't forget to revisit the implementation.